src/pages/core/architecture/actor-model.mdx

```diff
@@ -0,0 +1,58 @@
+---
+tags: ["architecture", "core"]
+---
+
+import Image from "next/image";
+import { Callout } from "nextra/components";
+
+# Actor Model
+
+CosmWasm, at its core, is built around the actor model. This prevents some
+common pitfalls which, for example, Ethereum smart contracts had to fix over the
+years. The issues mostly revolved around
+[reentrancy issues](https://ethereum.org/en/developers/docs/smart-contracts/security/#reentrancy)
+where we could, for example, call out to another contract before finishing the
+current execution.
+
+<Callout>
+  If you are already familiar with the actor model through, for example,
+  programming in languages such as Erlang or Elixir, you can skip this page.
+</Callout>
+
+While the term "actor model" sounds fancy, it is actually quite simple. Each
+contract is an "actor" and _plays_ a role in the system (pun intended).
+
+Each actor has its own state and can only be interacted with via messages. This
+means a contract can only interact with the outside world via messages and only
+manipulate its own state.
+
+Picture two people living in two houses, each house has a mailbox. If person A
+wants to talk to person B, they write a letter and put it in person B's mailbox.
+Person B will read the letter and write a response, putting it in person A's
+mailbox.
+
+No direct interaction is happening here, nobody visits each other's house which
+would allow them to rummage around in their belongings (state). They simply
+exchange messages and put everything the other side needs to know into a
+message.
+
+<Image
+  src="/actor-model.svg"
+  alt="Actor model"
+  style={{ marginTop: "1em", marginBottom: "1em" }}
+  width={1920}
+  height={1080}
+/>
+
+As you can see in the simple graphic above, they simply exchange messages
+addressed to each other, and the messages reference previous messages if need
+be.
+
+But how does that fix reentrancy? In CosmWasm, you can only send out messages at
+the end of a contract execution as part of the response. This ensures you have
+already written everything to the state, meaning your state is in a consistent
+state.
```

Comment:
"meaning your state is in a consistent state" is clever, but, could be confusing... maybe something like "meaning your state cannot be changed under your feet mid-execution", or something like that, e.g. explicitly point out the problem it's solving?